### PR TITLE
Added tests on calculated default value of properties depending on other properties

### PR DESCRIPTION
**Pull request by shanshin**
Tests for internal and external serializable classes. The default values for constructor properties, transient properties, body properties are checked

### CODE REVIEW

File: formats/json/commonTest/src/kotlinx/serialization/features/PropertyExternalInitializerTest.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+package kotlinx.serialization.features
+
+import kotlinx.serialization.*
+import kotlinx.serialization.json.Json
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+@Suppress("MemberVisibilityCanBePrivate", "unused", "ComplexRedundantLet")
+class PropertyExternalInitializerTest {
+    data class ExternalClass(
```
Comment (sandwwraith) [inline formats/json/commonTest/src/kotlinx/serialization/features/PropertyExternalInitializerTest.kt:12]: I think both these classes can be in the same file, I don't see any objections

Comment (shanshin) [reply formats/json/commonTest/src/kotlinx/serialization/features/PropertyExternalInitializerTest.kt:12]: Classes are separated by files to disable the compilation of external serializers in Legacy JS. Disable all initializer tests for the legacy JS?

Comment (sandwwraith) [reply formats/json/commonTest/src/kotlinx/serialization/features/PropertyExternalInitializerTest.kt:12]: Ah, OK then, leave them as is

Comment (shanshin) [reply formats/json/commonTest/src/kotlinx/serialization/features/PropertyExternalInitializerTest.kt]: Leave only one file - evaluated default value is unstable on legacy JS

**Review by sandwwraith — APPROVED** [review]

**Pull request merged**